tests/databrowser.js

```diff
@@ -200,6 +204,77 @@ define([
                 util.assertXmlEqual(call("createXML"), PRELOADED_HASHTAGS_XML, {normalize_xmlns: true});
             });
 
+            it("should add the casedb instance when referencing a case in a label", function(done) {
+                util.loadXML("");
+                var mug = util.addQuestion("Text", "mug"),
+                    label = $("[name=itext-en-label]"),
+                    sessionUri = CASE_DATA[0].uri,
+                    casedbUri = CASE_DATA[1].uri,
+                    editor = label.ckeditor().editor,
+                    widget = util.getWidget('itext-en-label');
+                widget.input.promise.then(function () { 
+                    editor.on('change', function() {
+                        assert.equal(mug.p.labelItext.get(), 'question1<output value="#case/child/dob" />');
+                        assert.equal(getInstanceId(mug.form, sessionUri), "commcaresession");
+                        assert.equal(getInstanceId(mug.form, casedbUri), "casedb");
+                        util.assertXmlEqual(call("createXML"), CHILD_REF_OUTPUT_VALUE_XML,
+                                            {normalize_xmlns: true});
+                        done();
+                    });
+                    assert.equal(getInstanceId(mug.form, sessionUri), null);
+                    assert.equal(getInstanceId(mug.form, casedbUri), null);
+                    assert.equal(label.length, 1);
+                    util.findNode(dataTree, "dob").data.handleDrop(label);
+                });
+            });
+
+            describe("with two languages", function () {
+                beforeEach(function (done) {
```

Comment:
non-blocking: curious why you recently started using `beforeEach` instead of `before`? I know it resets more state to re-init vellum for every test, but it also takes more time. My previous best practice was to use `before` in cases where most vellum state was unlikely to be changed by the tests in the group, and then `util.loadXML("");` (which is fast) to load a fresh form for each test. Should I be thinking about things differently?